src/Java/gtPlusPlus/xmod/gregtech/loaders/RecipeGen_DustGeneration.java

```diff
@@ -38,71 +38,72 @@ public RecipeGen_DustGeneration(final Material M, final boolean O){
 		this.toGenerate = M;		
 		this.disableOptional = O;
 		mRecipeGenMap.add(this);
-	}
-
-	@Override
-	public void run() {
-		generateRecipes(this.toGenerate, this.disableOptional);
-	}
-
-	private void generateRecipes(final Material material, final boolean disableOptional){
-
-		Logger.WARNING("Generating Shaped Crafting recipes for "+material.getLocalizedName());
-
-		final ItemStack normalDust = material.getDust(1);
-		final ItemStack smallDust = material.getSmallDust(1);
-		final ItemStack tinyDust = material.getTinyDust(1);
-
-		final ItemStack[] inputStacks = material.getMaterialComposites();
-		final ItemStack outputStacks = material.getDust(material.smallestStackSizeWhenProcessing);
-
-
-		if (ItemUtils.checkForInvalidItems(tinyDust) && ItemUtils.checkForInvalidItems(normalDust)) {
+		final ItemStack normalDust = M.getDust(1);
```

Comment:
Just curious: why do you move these to constructor?

Reply:
Because if I didn't they didn't work when the runnable was later called, no idea why.